CRM/Mosaico/Utils.php

```diff
@@ -326,12 +326,6 @@ public static function processImg() {
 
       switch ($method) {
         case 'placeholder':
-
-          // Only privileged users can request generation of placeholders
-          if (!CRM_Core_Permission::check([['access CiviMail', 'create mailings', 'edit message templates']])) {
-            CRM_Utils_System::permissionDenied();
-          }
-
```

Comment:
Did you check that the xml access_arguments mean we don't need the check here?

Reply:
Shouldn't require the check in php for sure.
I did test it for Joomla which seemed to work due to backend url login prompt. And I just tried it for drupal to confirm, and it wasn't working as expected. 
Unlike other urls in mosaico menu xml file, it seems to require a access_callback, may be because in the tree mosaico/img url is public, and is somehow being derived (as public).
Have explicitly set the access_callback and tested. Seems to work now. Don't need the check in php.
Thanks for the headsup.